sklearn/metrics/scorer.py

```diff
@@ -37,10 +38,15 @@ def __init__(self, score_func, sign, kwargs):
         self._kwargs = kwargs
         self._score_func = score_func
         self._sign = sign
+        # XXX After removing the deprecated scorers (v0.20) remove the
+        # XXX deprecation_msg property again and make __call__ abstract again
+        self._deprecation_msg = None
 
-    @abstractmethod
     def __call__(self, estimator, X, y, sample_weight=None):
-        pass
+        if self._deprecation_msg is not None:
+            warnings.warn(self._deprecation_msg,
+                          category=DeprecationWarning,
+                          stacklevel=13)
```

Comment:
Yes, it's always very hard to get informative line numbers for the user from warnings. I don't know why they don't store (at least, without printing) a traceback.